dataclients/kubernetes/ingress.go

```diff
@@ -351,6 +353,29 @@ func hasCatchAllRoutes(routes []*eskip.Route) bool {
 	return false
 }
 
+func addHostTLSCerts(ic ingressContext, routes []*eskip.Route, host string, secretName string) error {
+	for k := range routes {
+		for _, secret := range ic.state.secrets {
+			if secret.Meta.Name == secretName {
+				crt, err := b64.StdEncoding.DecodeString(secret.Data["tls.crt"])
+				if err != nil {
+					return err
+				}
+				key, err := b64.StdEncoding.DecodeString(secret.Data["tls.key"])
+				if err != nil {
+					return err
+				}
```

Comment:
* it should check the namespace as well there might be secrets with the same name in different namespaces. I think this part will be fixed if we change cluster state to use `secrets map[definitions.ResourceID]*secret` 
* it should check that secret is well-formed (here or maybe while reading the secret), i.e. check `secret.Type == "kubernetes.io/tls"` and presence of `tls.key` in the secret data (`StdEncoding.DecodeString` decodes empty string without error).

Reply:
Added some validations and warnings before converting secret to certificate. Still unclear how we would handle two secrets with the same name in different namespaces. Do we require the secret to be in the same namespace as the Ingress?

Reply:
Added a validation to make sure ingress and secret are in the same namespace.